app/src/screens/mapscreen.js

```diff
@@ -0,0 +1,282 @@
+          });
+          setLayers([highestSignalLayer, iconLayer]);
+          setMinRssiToDisplay(minRssi);
+          setMaxRssiToDisplay(maxRssi);
+        }
+      };
+      reader.readAsText(file);
+    });
+  }
+
+  function processFileData(allText) {
+    // Parse CSV with the header format:
+    // latitude, longitude, altitude, height, rssi, bearing, status, time
+    const allTextLines = allText.split(/\r\n|\n/);
+    const lines: Array<Point> = [];
+
+    for (let i = 1; i < allTextLines.length; i++) {
+      const data = allTextLines[i].split(',');
+      // Pull out the fields needed
+      const longitude: number = parseFloat(data[0]);
+      const latitude: number = parseFloat(data[1]);
+      const height: number = parseFloat(data[3]);
+      const rssi: number = parseFloat(data[4]);
+      const bearing: number = parseFloat(data[5]);
+
+      if (height > MIN_ELEVATION) {
+        if (typeof rssi === 'number' && rssi < maxRssi) {
+          maxRssi = rssi;
+        }
+        if (typeof rssi === 'number' && rssi > minRssi) {
+          minRssi = rssi;
+          bestLatitude = latitude;
+          bestLongitude = longitude;
+          bestElevation = height;
+          bestBearing = bearing;
+        }
+        lines.push({
+          latitude,
+          longitude,
+          height,
+          rssi,
+          bearing,
+          message:
+            longitude +
+            ', ' +
+            latitude +
+            ' ' +
+            height +
+            ' meters ' +
+            rssi +
+            'dBm ' +
+            bearing +
+            '\u00b0',
+        });
+      }
+    }
+    return lines;
+  }
+
+  function showMap() {
+    setMapStyle('mapbox://styles/mapbox/light-v9');
+    setSatelliteView(false);
+  }
+
+  function showSatellite() {
+    setMapStyle('mapbox://styles/mapbox/satellite-v9');
+    setSatelliteView(true);
+  }
+
+  return (
+    <div>
+      <DeckGL
+        initialViewState={view}
+        controller={true}
+        layers={layers}
+        getTooltip={(p: Point) => p.message}>
+        <ReactMapGL mapboxApiAccessToken={MAPBOX_TOKEN} mapStyle={mapStyle}>
+          <div style={{position: 'absolute', right: 0}}>
+            <NavigationControl showCompass={true} showZoom={false} />
+          </div>
+        </ReactMapGL>
+        <div
+          style={{
+            padding: 10,
+            background: 'rgba(255,255,255,0.8)',
+            display: 'inline-block',
+          }}>
+          <Button variant="contained" color="primary" onClick={handleOpenClick}>Open Files</Button>
+          <p />
+          Highest RSSI: {minRssiToDisplay}dBm
+          <p />
+          Lowest RSSI: {maxRssiToDisplay}dBm
+          <p />
+          <ButtonGroup>
+            <Button variant={satelliteView ? null : "contained"} color="primary" onClick={showMap}>Map</Button>
+            <Button variant={satelliteView ? "contained" : null} color="primary" onClick={showSatellite}>Satellite</Button>
```

Comment:
nit, break up to multiple lines to make it easier to read?
```suggestion
            <Button 
              variant={satelliteView ? null : "contained"}
              color="primary"
              onClick={showMap}>
              Map
            </Button>
            <Button
            variant={satelliteView ? "contained" : null}
            color="primary"
            onClick={showSatellite}>
            Satellite
          </Button>
```